### PR TITLE
Add health state and management server for coordinators

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

This diff adds docs for PR: https://github.com/memgraph/memgraph/pull/2138.

In the given PR management server coordinators is introduced, as well as health output is now the same for leaders and followers.

### Pull request type

Please check what kind of PR this is:

- [x] Fix or improvement of an existing page
- [ ] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
(especially necessary if the PR is related to a release)
 https://github.com/memgraph/memgraph/pull/2138

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

File: pages/clustering/high-availability.mdx [hunk 1]
```diff
@@ -36,11 +36,11 @@ One coordinator instance is the leader whose job is always to ensure one writeab
 own Raft log. Operations saved into the Raft log are those that are related to cluster management. Memgraph doesn't have its implementation of the Raft protocol. 
 For this task, Memgraph uses an industry-proven library [NuRaft](https://github.com/eBay/NuRaft).
 
-You can start the coordinator instance by specifying `--coordinator-id` and 
-`--coordinator-port` flags. The coordinator instance only responds to
+You can start the coordinator instance by specifying `--coordinator-id`, 
+`--coordinator-port` and `--management-port` flags. The `--management-port` is used on follower coordinators to get health state of cluster from leader coordinator. The coordinator instance only responds to
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:40]: What happens when a leader fails and is removed from the cluster, recovers, and then attempts to rejoin the cluster as a follower? The way this is written now, it will fail because it doesn't know the management port. Is that accurate?

File: pages/clustering/high-availability.mdx [hunk 2]
```diff
@@ -104,13 +104,15 @@ docker run --name coord1 -p 7690:7687 -p 7445:7444 memgraph/memgraph-mage
...
 ```
 
-Coordinator IDs serve as identifiers, while the coordinator port is used for synchronization and log replication between coordinators. Coordinator IDs and coordinator ports must be different for all coordinators.
+Coordinator IDs serve as identifiers, the coordinator port is used for synchronization and log replication between coordinators and management port is used to get health state of cluster from leader coordinator. 
+Coordinator IDs, coordinator ports and management ports must be different for all coordinators.
 
-Configuration option `--coordinator-hostname` must be set on all coordinator instances. It is used only in the `SHOW INSTANCES` query to correctly return networking information about
-the first leader (the coordinator instance on which you run registration queries). Setting this configuration flag to some random value will not cause issues in the cluster's capabilities
-but you will get misleading output when running `SHOW INSTANCES` query. 
+Configuration option `--coordinator-hostname` must be set on all coordinator instances. It is used on followers to ping leader coordinator on correct IP address and return health state about
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:113]: s/ping leader/ping the leader
s/on correct/on the correct

File: pages/clustering/high-availability.mdx [hunk 3]
```diff
@@ -267,21 +270,13 @@ This operation will result in writing to the Raft log.
 You can check the state of the whole cluster using the `SHOW INSTANCES` query. The query will show which instances are visible in the cluster,
 which network ports they are using for managing cluster state, whether they are considered alive from the coordinator's
 perspective, their role (are they MAIN, REPLICA, LEADER, FOLLOWER or unknown if not alive) and the time elapsed in ms since the last successful response
-from instances to the leader's ping. This query can be run both on leader and followers with the sole difference that followers cannot know the 
-health status of other instances and the time of last response since only the leader is pinging instances for their status.
+from instances to the leader's ping. This query can be run both on leader and followers. Followers will try to contact leader to get state of the cluster.
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:273]: s/contact leader/contact the leader
s/get state/get the state

File: pages/clustering/high-availability.mdx [hunk 4]
```diff
@@ -267,21 +270,13 @@ This operation will result in writing to the Raft log.
 You can check the state of the whole cluster using the `SHOW INSTANCES` query. The query will show which instances are visible in the cluster,
 which network ports they are using for managing cluster state, whether they are considered alive from the coordinator's
 perspective, their role (are they MAIN, REPLICA, LEADER, FOLLOWER or unknown if not alive) and the time elapsed in ms since the last successful response
-from instances to the leader's ping. This query can be run both on leader and followers with the sole difference that followers cannot know the 
-health status of other instances and the time of last response since only the leader is pinging instances for their status.
+from instances to the leader's ping. This query can be run both on leader and followers. Followers will try to contact leader to get state of the cluster. 
+In case contacting leader fails, followers will return data with time elapsed in ms since the last successful response set to 0, and health state unknown.
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:274]: s/contacting leader/contacting the leader
s/with time/with the time

And we need to better indicate that the phrase "time elapsed in ms since the last successful response" is one specific thing. It's a long phrase and hard to parse.

File: pages/clustering/high-availability.mdx (hunk 1 above)
Comment (antoniofilipovic) [reply pages/clustering/high-availability.mdx:40]: @kgolubic that is not fully correct, it is not removed from the cluster, and it will not ping the leader to get a management port.

File: pages/clustering/high-availability.mdx (hunk 4 above)
Comment (antoniofilipovic) [reply pages/clustering/high-availability.mdx:274]: Agreed, changed phrase to `response time`

File: pages/clustering/high-availability.mdx (hunk 1 above)
Comment (antoniofilipovic) [reply pages/clustering/high-availability.mdx]: I changed the wording a bit, let me know if you have additional comments

**Review by kgolubic — APPROVED** [review]
It all looks ok to me.

File: pages/clustering/high-availability.mdx [hunk 5]
```diff
@@ -104,13 +104,15 @@ docker run --name coord1 -p 7690:7687 -p 7445:7444 memgraph/memgraph-mage
...
+Coordinator IDs serve as identifiers, the coordinator port is used for synchronization and log replication between coordinators and management port is used to get health state of cluster from leader coordinator. 
+Coordinator IDs, coordinator ports and management ports must be different for all coordinators.
 
-Configuration option `--coordinator-hostname` must be set on all coordinator instances. It is used only in the `SHOW INSTANCES` query to correctly return networking information about
-the first leader (the coordinator instance on which you run registration queries). Setting this configuration flag to some random value will not cause issues in the cluster's capabilities
-but you will get misleading output when running `SHOW INSTANCES` query. 
+Configuration option `--coordinator-hostname` must be set on all coordinator instances. It is used on followers to ping the leader coordinator on the correct IP address and return health state about
+the cluster. Setting this configuration flag to some random value will cause issues in the cluster's capabilities
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:114]: Before talking about random values, let's give some examples of successful values. Does it need to be an ip address? an FQDN? can it appear in the /etc/hosts file?

Comment (antoniofilipovic) [reply pages/clustering/high-availability.mdx]: I changed it, let me know if it is not fine

File: pages/clustering/high-availability.mdx [hunk 6]
```diff
@@ -266,22 +268,14 @@ This operation will result in writing to the Raft log.
 
 You can check the state of the whole cluster using the `SHOW INSTANCES` query. The query will show which instances are visible in the cluster,
 which network ports they are using for managing cluster state, whether they are considered alive from the coordinator's
-perspective, their role (are they MAIN, REPLICA, LEADER, FOLLOWER or unknown if not alive) and the time elapsed in ms since the last successful response
-from instances to the leader's ping. This query can be run both on leader and followers with the sole difference that followers cannot know the 
-health status of other instances and the time of last response since only the leader is pinging instances for their status.
+perspective, their role (are they MAIN, REPLICA, LEADER, FOLLOWER or unknown if not alive) and the response time
+from instances to the leader's ping. This query can be run both on the leader and followers. Followers will try contacting the leader to get the state of the cluster. 
+If contacting the leader fails, followers will return time since the last successful response and health state set to 0 and unknown, respectfully.
```
Comment (hal-eisen-MG) [inline pages/clustering/high-availability.mdx:273]: I think you mean "respectively". However, I'm still having trouble understanding what happens when contacting the leader fails. Perhaps make an ordered list to clearly separate the results?

Comment (antoniofilipovic) [reply pages/clustering/high-availability.mdx:273]: Can you explain which part is problematic, I have trouble fixing it if I am not sure what the exact issue is with understanding :smile: 
I will rewrite what is happening, but I think a better way on commenting a sentence would be to say which part exactly is not understood

Comment (kgolubic) [reply pages/clustering/high-availability.mdx]: Note: the [development PR](https://github.com/memgraph/memgraph/pull/2138) is marked as `breaking`. Due to that, if all comments are not resolved until the moment we release 2.18.1, I'll create new issues to resolve the remainder of the open issues after the release.
